DigitalLearningSolutions.Web.Tests/ViewModels/TrackingSystem/Centre/Administrator/EditRolesViewModelTests.cs

```diff
@@ -76,12 +78,246 @@ public void EditRolesViewModel_sets_ContentManagementRole_none()
+        {
+            // Given
+            var adminUser = UserTestHelper.GetDefaultAdminUser(importOnly: false, isContentManager: false);
+            var numberOfAdmins =
+                NumberOfAdministratorsTestHelper.GetDefaultNumberOfAdministrators(cmsManagers: 5, cmsManagerSpots: 5);
+
+            // When
+            var result = new EditRolesViewModel(adminUser, 1, categories, numberOfAdmins);
+
+            // Then
+            using (new AssertionScope())
+            {
+                result.Radios.Count.Should().Be(2);
+                result.Radios.Contains(AdminRoleInputs.CmsManagerRadioButton).Should().BeFalse();
+                result.NotAllRolesDisplayed.Should().BeTrue();
+                result.NoContentManagerOptionsAvailable.Should().BeFalse();
+            }
+        }
+
+        [Test]
+        public void
+            EditRolesViewModel_does_set_up_Cms_manager_radio_button_when_its_limit_is_reached_and_user_is_a_cms_manager()
+        {
+            // Given
+            var adminUser = UserTestHelper.GetDefaultAdminUser(importOnly: false, isContentManager: true);
+            var numberOfAdmins =
+                NumberOfAdministratorsTestHelper.GetDefaultNumberOfAdministrators(cmsManagers: 5, cmsManagerSpots: 5);
+
+            // When
+            var result = new EditRolesViewModel(adminUser, 1, categories, numberOfAdmins);
+
+            // Then
+            using (new AssertionScope())
+            {
+                result.Radios.Count.Should().Be(3);
+                result.Radios.Contains(AdminRoleInputs.CmsManagerRadioButton).Should().BeTrue();
+                result.NotAllRolesDisplayed.Should().BeFalse();
+                result.NoContentManagerOptionsAvailable.Should().BeFalse();
+            }
+        }
+
+        [Test]
+        public void
+            EditRolesViewModel_sets_only_NoCmsPermissions_radio_button_when_both_limits_reached_and_user_has_no_permissions()
+        {
+            // Given
+            var adminUser = UserTestHelper.GetDefaultAdminUser(importOnly: false, isContentManager: false);
+            var numberOfAdmins =
+                NumberOfAdministratorsTestHelper.GetDefaultNumberOfAdministrators(
+                    cmsManagers: 5,
+                    cmsManagerSpots: 5,
+                    cmsAdministrators: 5,
+                    cmsAdministratorSpots: 5
+                );
+
+            // When
+            var result = new EditRolesViewModel(adminUser, 1, categories, numberOfAdmins);
+
+            // Then
+            using (new AssertionScope())
+            {
+                result.Radios.Count.Should().Be(1);
+                result.Radios.Contains(AdminRoleInputs.CmsManagerRadioButton).Should().BeFalse();
+                result.Radios.Contains(AdminRoleInputs.CmsAdministratorRadioButton).Should().BeFalse();
+                result.Radios.Contains(AdminRoleInputs.NoCmsPermissionsRadioButton).Should().BeTrue();

```

Comment:
If we check the count = 1 and "contains", do we need to check that it doesn't contain bunch of values?